src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -234,6 +235,8 @@ internal unsafe QuicConnection(QUIC_HANDLE* handle, QUIC_NEW_CONNECTION_INFO* in
 
     private async ValueTask FinishConnectAsync(QuicClientConnectionOptions options, CancellationToken cancellationToken = default)
     {
+        ObjectDisposedException.ThrowIf(_disposed == 1, this);
```

Comment:
BTW I still feel this is dead code and should not be here. But that can be addresses separately.

Reply:
This was bad merge. I merged some files manually from an older branch and must have missed it. I removed it and looked for other places as well.